### PR TITLE
Dictionary from OSM_id to its geojson linestring

### PR DESCRIPTION
**Pull request by CasperKjaerhus**
Use `get_osmid_to_linestring_dictionary` with a osm.pbf file. 
Theres a file on our shared drive under Data that you can use.

I'm a bit confused on how i can test this function without going hardcore into mocking a osm.pbf file

Solves issues #18

### CODE REVIEW

File: open_street/osm_geometry.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        if w.tags.get("highway") is not None:
+            try:
+                geo = GeoJSONFactory().create_linestring(w)  # Get the road linestring
+                self.geometryDictionary[w.id] = geo
+            except Exception as e:
+                print("error", e)
+                return
+
+
+def geometry_dictionary_to_geojson(geoDict: dict[str, str]) -> str:
```
Comment (AlexanderHHansen) [inline open_street/osm_geometry.py:26]: hvis geoDict maper fra id til dict af attributes, så skal typehint vel være dict[str,dict]?

File: open_street/osm_geometry.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    # Remove last comma since we are finished with the array
+    featureCollecton = featureCollecton.rstrip(featureCollecton[-1])
+
+    # End the geoJson string
+    featureCollecton += "]}"
+
+    return featureCollecton
+
+
+def get_osmid_to_linestring_dictionary(OSMFilePath: str) -> dict[str, str]:
```
Comment (AlexanderHHansen) [inline open_street/osm_geometry.py:52]: skal typehint være dict[str,dict] ?

File: open_street/osm_geometry.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import osmium
+from osmium.geom import GeoJSONFactory
+from typing import Any
+
+
+class GeometryRoadHandler(osmium.SimpleHandler):
+    def __init__(self) -> None:
+        osmium.SimpleHandler.__init__(self)
+        self.geometryDictionary: dict[str, str] = {}
```
Comment (AlexanderHHansen) [inline open_street/osm_geometry.py:9]: skal typehint være dict[str,dict] ?

Comment (CasperKjaerhus) [reply open_street/osm_geometry.py:9]: Nej det er en json formatted string der kommer ind :)

File: open_street/osm_geometry.py (hunk 1 above)
Comment (CasperKjaerhus) [reply open_street/osm_geometry.py:26]: Den mapper udelukket til geojson string

**Review by AlexanderHHansen — APPROVED** [review]

File: open_street/osm_geometry.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    def way(self, w: Any) -> None:
+        """This function gets called upon .apply_file for every road (OSM Way) that is in the file that apply_file is called with.
+
+        Args:
+            w (Any): OSM Way
+        """
+        if w.tags.get("highway") is not None:
```
Comment (frejaz) [inline open_street/osm_geometry.py:17]: # The highway tag annotates the type of road, e.g. 'path'

File: open_street/osm_geometry.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+from typing import Any
+
+
+class GeometryRoadHandler(osmium.SimpleHandler):
+    def __init__(self) -> None:
+        osmium.SimpleHandler.__init__(self)
+        self.geometryDictionary: dict[str, str] = {}
+
+    def way(self, w: Any) -> None:
+        """This function gets called upon .apply_file for every road (OSM Way) that is in the file that apply_file is called with.
```
Comment (frejaz) [inline open_street/osm_geometry.py:14]: Synes det her lyder lidt convoluted. Idea:
"Way() looks for all ways with the tag "highway" and adds their osm id and linestring to the geometryDictionary. This function is used by apply_file(), which is method inherited by osmium.SimpleHandler - the handler will look for a function called way() and apply the function on each OSM way (road) in a given .pbf file."

File: open_street/osm_geometry.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+            except Exception as e:
+                print("error", e)
+                return
+
+
+def geometry_dictionary_to_geojson(geoDict: dict[str, str]) -> str:
+    """outputs geoJson formatted string from at osm_id -> linestring dictionary
+
+    Args:
+        geoDict (dict[str, str]): A dictionary that maps osm_ids to their geojson dictionary
```
Comment (frejaz) [inline open_street/osm_geometry.py:34]: Måske bare add "... to their geojson dictionary (formated as string)"

File: open_street/osm_geometry.py [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+            try:
+                geo = GeoJSONFactory().create_linestring(w)  # Get the road linestring
+                self.geometryDictionary[w.id] = geo
+            except Exception as e:
+                print("error", e)
+                return
+
+
+def geometry_dictionary_to_geojson(geoDict: dict[str, str]) -> str:
+    """outputs geoJson formatted string from at osm_id -> linestring dictionary
```
Comment (frejaz) [inline open_street/osm_geometry.py:31]: Sjovt at du starter med outputtet, jeg havde nok flippet det - "Convert geometryDictionary to geoJSON string"

File: open_street/osm_geometry.py (hunk 6 above)
Comment (CasperKjaerhus) [reply open_street/osm_geometry.py:34]: ye er kommet til at skrive forkert i kommentaren kan jeg se :))))))))))))))))))))))))))))))))))))))))))))))))9

File: open_street/osm_geometry.py (hunk 5 above)
Comment (CasperKjaerhus) [reply open_street/osm_geometry.py]: Nice! har rettet (mini) i det :)

**Review by frejaz — APPROVED** [review]
Nice!

**Pull request merged**